weld/codegen/llvm2/cmp.rs

```diff
@@ -0,0 +1,351 @@
+                let left_value = self.gen_at(builder, ty, left_vector, phi_i)?;
+                let right_value = self.gen_at(builder, ty, right_vector, phi_i)?;
+                let mut args = [left_value, right_value];
+                let cmp_result = LLVMBuildCall(builder, func, args.as_mut_ptr(), args.len() as u32, c_str!(""));
+                let neq = LLVMBuildICmp(builder, LLVMIntNE, cmp_result, self.i32(0), c_str!(""));
+
+                let updated_i = LLVMBuildNSWAdd(builder, phi_i, self.i64(1), c_str!(""));
+                let check1 = LLVMBuildICmp(builder, LLVMIntEQ, updated_i, left_size, c_str!(""));
+                let check2 = LLVMBuildOr(builder, check1, neq, c_str!(""));
+                // End loop if i == size || left != right
+                LLVMBuildCondBr(builder, check2, done_block, loop_block);
+
+                let mut blocks = [entry_block, loop_block];
+                let mut values = [self.i64(0), updated_i];
+                LLVMAddIncoming(phi_i, values.as_mut_ptr(), blocks.as_mut_ptr(), values.len() as u32);
+
+                // Finish block - set result and compute number of consumed bits.
+                LLVMPositionBuilderAtEnd(builder, done_block);
+                let result = LLVMBuildPhi(builder, self.i32_type(), c_str!(""));
+                let mut blocks = [entry_block, loop_block];
+                let mut values = [self.i32(1), cmp_result];
+                LLVMAddIncoming(result, values.as_mut_ptr(), blocks.as_mut_ptr(), values.len() as u32);
+
+                LLVMBuildRet(builder, result);
+            }
+            Function(_,_) | Unknown => {
+                unreachable!()
+            }
+        };
+
+        LLVMDisposeBuilder(builder);
+
+        self.cmp_fns.insert(ty.clone(), function);
+        Ok(function)
+    }
+
+    /// Generates an opaque comparison function for a type.
+    unsafe fn gen_opaque_cmp_fn(&mut self, ty: &Type) -> WeldResult<LLVMValueRef> {
+        let result = self.opaque_cmp_fns.get(ty).cloned();
+        if let Some(result) = result {
+            return Ok(result)
+        }
+
+        let llvm_ty = self.llvm_type(ty)?;
+        let mut arg_tys = [self.void_pointer_type(), self.void_pointer_type()];
+        let ret_ty = self.i32_type();
+
+        let c_prefix = LLVMPrintTypeToString(llvm_ty);
+        let prefix = CStr::from_ptr(c_prefix);
+        let prefix = prefix.to_str().unwrap();
+        let name = format!("{}.opaque_cmp", prefix);
+
+        // Free the allocated string.
+        LLVMDisposeMessage(c_prefix);
+
+        let (function, builder, _) = self.define_function_with_visibility(ret_ty,
+                                                                          &mut arg_tys,
+                                                                          LLVMLinkage::LLVMExternalLinkage,
+                                                                          name);
+
+        LLVMExtAddAttrsOnFunction(self.context, function, &[InlineHint]);
```

Comment:
Delete this since it shouldn't be inlined (it's only purpose is to be passed to a libc function)